process/block/shardblock.go

```diff
@@ -650,11 +652,24 @@ func (sp *shardProcessor) CommitBlock(
 		log.Debug(errNotCritical.Error())
 	}
 
-	errNotCritical = sp.forkDetector.AddHeader(header, headerHash, process.BHProcessed)
+	finalHeader, errNotCritical := sp.getHighestHdrForOwnShardFromMetachain(header.Round)
```

Comment:
you could put finalHeaderHsah calculation inside getHighestHdrForOwnShardFromMetachain function.

Reply:
Done